pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/design/AvoidDeeplyNestedIfStmtsRule.java

```diff
@@ -4,20 +4,24 @@
 
 package net.sourceforge.pmd.lang.apex.rule.design;
 
+import static net.sourceforge.pmd.properties.constraints.NumericConstraints.positive;
+
 import net.sourceforge.pmd.lang.apex.ast.ASTIfBlockStatement;
 import net.sourceforge.pmd.lang.apex.ast.ASTUserClass;
 import net.sourceforge.pmd.lang.apex.rule.AbstractApexRule;
-import net.sourceforge.pmd.properties.IntegerProperty;
+import net.sourceforge.pmd.properties.PropertyDescriptor;
+import net.sourceforge.pmd.properties.PropertyFactory;
+
 
 public class AvoidDeeplyNestedIfStmtsRule extends AbstractApexRule {
 
     private int depth;
     private int depthLimit;
 
-    private static final IntegerProperty PROBLEM_DEPTH_DESCRIPTOR
-            = IntegerProperty.named("problemDepth")
+    private static final PropertyDescriptor<Integer> PROBLEM_DEPTH_DESCRIPTOR
+            = PropertyFactory.intProperty("problemDepth")
                              .desc("The if statement depth reporting threshold")
-                             .range(1, 25).defaultValue(3).uiOrder(1.0f).build();
+                             .require(positive()).defaultValue(3).build();
```

Comment:
Just wondered, whether we would have documented the minimum/maximum already (which we now change....) - but we don't: https://pmd.github.io/pmd-6.9.0/pmd_rules_apex_design.html#avoiddeeplynestedifstmts
So. no problem here. Limiting the the problemDepth to 25 was arbitrary anyways.